cmd/networkPolicies.go

```diff
@@ -3,35 +3,82 @@ package cmd
 import (
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
+	networking "k8s.io/api/networking/v1"
 )
 
-func checkNamespaceNetworkPolicies(netPols *NetworkPolicyList) {
-	badNetPols := []NetworkPolicy{}
+func getNamespacesMap(namespaceList *NamespaceList) map[string]bool {
+	// TODO filter namespaces based on annotations
+	nsMap := map[string]bool{}
+
+	for _, ns := range namespaceList.Items {
+		nsMap[ns.Name] = false
+	}
+
+	return nsMap
+}
+
+func printResult(auditedNamspaces map[string]bool) {
+	for ns, hasDefaultDenyNetPol := range auditedNamspaces {
+		if hasDefaultDenyNetPol {
+			log.WithField("KubeType", "netpol").WithField("Namespace", ns).Info("Has default deny NeworkPolicy isolation")
+		} else {
+			log.WithField("KubeType", "netpol").WithField("Namespace", ns).Error("Missing default deny NeworkPolicy isolation")
+		}
+	}
+}
+
+// Currently only checks ingress rules
+func checkIfDefaultDenyPolicy(netpol networking.NetworkPolicy) bool {
+	// No PodSelector is set via MatchLabels -> Catch all pods
+	if len(netpol.Spec.PodSelector.MatchLabels) > 0 {
+		return false
+	}
+
+	// No PodSelector is set via MatchExpressions -> catch all Pods
+	if len(netpol.Spec.PodSelector.MatchExpressions) > 0 {
+		return false
+	}
+
+	// No Ingress rule is defined -> denie all ingress traffic
+	if len(netpol.Spec.Ingress) > 0 {
+		return false
+	}
+
+	// TODO check if ingress or egress rule? if netpol.Spec.PolicyTypes
+	// TODO evalute also egress?
```

Comment:
Since these are some small changes I would add them here. So a `default-deny` policy would prevent any incoming and outgoing traffic correct? So we expect a rule like this: https://kubernetes.io/docs/concepts/services-networking/network-policies/#default-deny-all-ingress-and-all-egress-traffic ?